src/fly/sorts.rs

```diff
@@ -19,100 +20,100 @@ pub enum SortError {
 
     #[error("expected {0} but found {1}")]
     NotEqual(Sort, Sort),
+    #[error("expected {0} args but found {1} args")]
+    ArgMismatch(usize, usize),
```

Comment:
For this variant and the above I'd highly recommend using a struct, to avoid swapping the arguments. It would look like `ArgMismatch { expected: usize, found: usize }`.

Reply:
Hmm, in the code there isn't really a way to differentiate them. Maybe it'd be better to change the error message instead? I'm not sure quite what to change it to though.

Reply:
For `ArgMismatch` I believe it really is expected followed by found (because expected comes from the inferred type of the function). You should probably also includes the function name there.

For `NotEqual` I understand your concern. Changing the error to `could not unify {0} and {1}` seems like a good solution; it does seem confuse to label them "expected" and "found" when as far as we know so far they're arbitrary.

Reply:
I think you thought that about ArgMismatch because I had a bug. I've changed the name to NotEqualArgs and added it into sort_eq like it should have been.